benchmark/Benchmark.java

```diff
@@ -1,56 +1,117 @@
+    private int batchSize;
 
-    public Benchmark(int writePercentage, Database database, int numberOfNodes, Dataset dataset) {
+    public Benchmark(int writePercentage, Database[] databases, int numberOfNodes, Dataset dataset, int numberOfQueries, int batchSize) {
         this.writePercentage = writePercentage;
-        this.database = database;
+        this.databases = databases;
         this.numberOfNodes = numberOfNodes;
         this.dataset = dataset;
+        this.numberOfQueries = numberOfQueries;
+        this.batchSize = batchSize;
+
+    }
+
+    public String[][] generateQueryWorkload() {
+        String[][] queries = new String[databases.length][numberOfQueries];
+        Random random = new Random();
+        for (int i = 0; i < numberOfQueries; i++) {
+            int x = random.nextInt(100);
+
+            //write query
+            if (x < writePercentage) {
+                String[][] data = generateData(batchSize);
+                for (int j=0; j<databases.length; j++){
+                    queries[j][i] = databases[j].getQueryTranslator().translateInsertInto(dataset, data);
+                }
+            }
+            //read query
+            else {
+                QueryType type = QueryType.values()[random.nextInt(QueryType.values().length)];
+                for (int j=0; j<databases.length; j++) {
+                    queries[j][i] = generateQuery(databases[j].getQueryTranslator(), type);
+                }
+            }
+        }
+        return queries;
+    }
+
+    private String generateQuery(QueryTranslator queryTranslator, QueryType type) {
+        switch (type){
+            case EXACT_POINT -> {
+                //return queryTranslator.translateExactPoint();
+            }
+            case RANGE_ANY_ENTITY -> {
+                //
+            }
+            // TODO
+        }
+        return "";
+    }
+
+    private String[][] generateData(int batchSize) {
+        // generates batchSize many records that match the dataset given as a class variable
+        List<Map.Entry<String, ColumnType>> columns = dataset.getColumnNamesWithTypes();
+        String[][] data = new String[batchSize][columns.size()];
+        for (int i=0; i<batchSize; i++){
+            data[i] = generateRow(columns);
+        }
+        return data;
+    }
 
-        this.database.setup(this.dataset);
-        this.queryTranslator = database.getQueryTranslator();
+    private String[] generateRow(List<Map.Entry<String, ColumnType>> columns) {
+        Random random = new Random();
+        String[] row = new String[columns.size()];
+        for (int i=0; i<columns.size(); i++) {
+            switch (columns.get(i).getValue()){
+                case TIMESTAMP -> row[i] = Timestamp.from(Instant.now()).toString();
+                case NUMERIC -> row[i] = String.valueOf(random.nextDouble());
+                case STRING -> row[i] = java.util.UUID.randomUUID().toString();
+            }
+        }
+        return row;
```

Comment:
maybe let's move it to separate module and create static class `DataGenerator`? if I remember correctly that was our first idea. Also it will be easier to handle when code grows

Reply:
I added class DataGenerator now